browser-test/src/address.test.ts

```diff
@@ -0,0 +1,248 @@
+      error = await pageObject.$('.cf-address-city-error >> nth=1');
+      expect(await error.isHidden()).toEqual(false);
+      error = await pageObject.$('.cf-address-state-error >> nth=1');
+      expect(await error.isHidden()).toEqual(false);
+      error = await pageObject.$('.cf-address-zip-error >> nth=1');
+      expect(await error.isHidden()).toEqual(false);
+    });
+  });
+
+  // One optional address followed by one required address.
+  describe('optional address question', () => {
+    let applicantQuestions;
+    const programName = 'test program for optional address';
+
+    beforeAll(async () => {
+      await loginAsAdmin(pageObject);
+      const adminQuestions = new AdminQuestions(pageObject);
+      const adminPrograms = new AdminPrograms(pageObject);
+      applicantQuestions = new ApplicantQuestions(pageObject);
+
+      await adminQuestions.addAddressQuestion({questionName: 'address-test-optional-q'});
+      await adminQuestions.addAddressQuestion({questionName: 'address-test-required-q'});
+      await adminPrograms.addProgram(programName);
+      await adminPrograms.editProgramBlockWithOptional(programName, 'Optional question block', ['address-test-required-q'], 'address-test-optional-q');
+      await adminPrograms.gotoAdminProgramsPage();
+      await adminPrograms.publishAllPrograms();
+
+      await logout(pageObject);
+    });
+
+    it('with valid required address does submit', async () => {
+      await loginAsGuest(pageObject);
+      await selectApplicantLanguage(pageObject, 'English');
+
+      await applicantQuestions.applyProgram(programName);
+      await applicantQuestions.answerAddressQuestion('1234 St', 'Unit B', 'Sim', 'Ames', '54321', 1);
+      await applicantQuestions.clickNext();
+
+      await applicantQuestions.submitFromReviewPage(programName);
+    });
+
+    describe('with invalid required address', () => {
+      beforeEach(async () => {
+        await loginAsGuest(pageObject);
+        await selectApplicantLanguage(pageObject, 'English');
+
+        await applicantQuestions.applyProgram(programName);
+        await applicantQuestions.answerAddressQuestion('', '', '', '', '', 1);
+        await applicantQuestions.clickNext();
+
+      });
+
+      it('does not submit', async () => {
+        // Second question hos errors.
+        let error = await pageObject.$('.cf-address-street-1-error >> nth=1');
+        expect(await error.isHidden()).toEqual(false);
+        error = await pageObject.$('.cf-address-city-error >> nth=1');
+        expect(await error.isHidden()).toEqual(false);
+        error = await pageObject.$('.cf-address-state-error >> nth=1');
+        expect(await error.isHidden()).toEqual(false);
+        error = await pageObject.$('.cf-address-zip-error >> nth=1');
+        expect(await error.isHidden()).toEqual(false);
+      });
+
+      // TODO: Fix code so this test passes.
+      /*
+      it('optional has no errors', async () => {
```

Comment:
I thought optional only shows errors if the required one does. This one fails with optional being the only empty question?

Reply:
This is the case where if the required one has an error, the empty-optional shows error messages.  

Are you saying that's WAI even though optionals don't have to be specified?

Reply:
Oops, I misread it - empty-optional shouldn't show errors in this case. We can uncomment it once https://github.com/seattle-uat/civiform/issues/1652 is resolved.

Reply:
added issue references in the comment.